go/vt/tabletserver/gorpcqueryservice/sqlquery.go

```diff
@@ -23,13 +23,40 @@ type SqlQuery struct {
 // GetSessionId is exposing tabletserver.SqlQuery.GetSessionId
 func (sq *SqlQuery) GetSessionId(sessionParams *proto.SessionParams, sessionInfo *proto.SessionInfo) (err error) {
 	defer sq.server.HandlePanic(&err)
-	return sq.server.GetSessionId(sessionParams, sessionInfo)
+	tErr := sq.server.GetSessionId(sessionParams, sessionInfo)
+	tabletserver.AddTabletErrorToSessionInfo(tErr, sessionInfo)
+	if *tabletserver.RPCErrorOnlyInReply {
+		return nil
+	}
+	return tErr
 }
 
 // Begin is exposing tabletserver.SqlQuery.Begin
 func (sq *SqlQuery) Begin(ctx context.Context, session *proto.Session, txInfo *proto.TransactionInfo) (err error) {
```

Comment:
We also need new "Begin" RPC accepting BeginRequest and returning BeginResponse.